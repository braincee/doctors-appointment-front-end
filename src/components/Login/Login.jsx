import React from 'react'

const Login = () => {
  return (
    <div className="container-fluid w-50 mx-auto mt-5"> 
      <h2 className="h2 mt-5 mb-3 text-center">Log in</h2>
      <form>
        
        <div className="form-floating mb-3">
          <input
            type="text"
            className="form-control"
            id="username"
            placeholder="Username"
            value={username}
            required
          />
          <label htmlFor="username">Username</label>
        </div>
        <button type="submit" className="btn btn-primary">
          Log in
        </button>
      </form>
    </div>
  )
}

export default Login