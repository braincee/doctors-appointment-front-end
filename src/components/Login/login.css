.login-section {
  background-image: url("../assets/login.jpg");
  background-size: cover;
  height: 100vh;
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
}

.login-container {
  background-color: #b4b2ae3f;
  -webkit-backdrop-filter: blur(2px);
  backdrop-filter: blur(1px);
  padding: 80px;
  border-radius: 10px;
}

.button1:hover {
  color: #000000e5;
  background-color: #555453b4;
  font-family: 'Times New Roman', Times, serif;
}

.login-title {
  text-align: center;
  color: rgba(129, 124, 116, 0.986);
  margin-bottom: 40px;
}

.login-form {
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
  gap: 65px;
  margin-bottom: 5px;
}

form input {
  padding: 10px;
  border: 1px hidden;
  border-radius: 5px;
}
