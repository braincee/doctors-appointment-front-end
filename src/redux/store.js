import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';




const initialState = {};
const middleware = [thunk];

const store = createStore(initialState, (applyMiddleware(...middleware)));