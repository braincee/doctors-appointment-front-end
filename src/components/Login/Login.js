/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { NavLink, useNavigate } from 'react-router-dom';
import { userLogin } from '../../redux/user/login';
import './login.css';

const Login = () => {
  const [name, setName] = useState('');
  const [errors, setErrors] = useState('');
  const [success, setSuccess] = useState('');

  const dispatch = useDispatch();
  const navigate = useNavigate();

  const newState = useSelector((state) => state.loginSessionsReducer);

  const handleChange = (e) => {
    const input = e.target;
    const newUsername = input.value;
    switch (input.name) {
      case 'name':
        setName(newUsername);
        return name;

      default:
        return 'yes';
    }
  };

  const handleSubmit = (e) => {
    const form = e.target;
    e.preventDefault();
    form.reset();

    dispatch(userLogin(name));

    if (newState.status === 200) {
      setSuccess(newState.fetchedData.message);
      navigate('/doctors');
    } else {
      setErrors(newState.fetchedData.error);
    }
  };

  return (
    <section className="login-section">
      <div className="login-container">
        <h2 className="login-title">Login</h2>
        <form onSubmit={handleSubmit} className="login-form">
          <input
            type="text"
            name="name"
            value={name}
            placeholder="Enter your name"
            onChange={handleChange}
            required
          />
          <div className="d-flex">
            <button type="submit" className=" button1 btn btn-secondary me-4 fs-4">
              Log in
            </button>
            <NavLink to="/signup">
              <button type="button" className="button1 btn btn-secondary me-4 fs-4">Sign Up</button>
            </NavLink>
          </div>
          <>
            <p>{success}</p>
            <p>{errors}</p>
          </>
        </form>
      </div>
    </section>
  );
};

export default Login;
