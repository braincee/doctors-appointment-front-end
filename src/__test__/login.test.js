import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Login from '../components/Login/Login';
import store from '../redux/configureStore';

describe('Log in tests', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Log in page has log in button', () => {
    const login = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <Login />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(login.findByText('Log in')).toBeTruthy();
  });

  it('Log in page has a sign up link', () => {
    const login = render(
      <>
        <Provider store={store}>
          <BrowserRouter>
            <Login />
          </BrowserRouter>
        </Provider>
      </>,
    );

    expect(login.findByText('Sign Up')).toBeTruthy();
  });
});
