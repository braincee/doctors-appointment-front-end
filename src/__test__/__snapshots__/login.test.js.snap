// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log in tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="login-section"
      >
        <div
          class="login-container"
        >
          <h2
            class="login-title"
          >
            Login
          </h2>
          <form
            class="login-form"
          >
            <input
              name="name"
              placeholder="Enter your name"
              required=""
              type="text"
              value=""
            />
            <div
              class="d-flex"
            >
              <button
                class=" button1 btn btn-secondary me-4 fs-4"
                type="submit"
              >
                Log in
              </button>
              <a
                class=""
                href="/signup"
              >
                <button
                  class="button1 btn btn-secondary me-4 fs-4"
                  type="button"
                >
                  Sign Up
                </button>
              </a>
            </div>
            <p />
          </form>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="login-section"
    >
      <div
        class="login-container"
      >
        <h2
          class="login-title"
        >
          Login
        </h2>
        <form
          class="login-form"
        >
          <input
            name="name"
            placeholder="Enter your name"
            required=""
            type="text"
            value=""
          />
          <div
            class="d-flex"
          >
            <button
              class=" button1 btn btn-secondary me-4 fs-4"
              type="submit"
            >
              Log in
            </button>
            <a
              class=""
              href="/signup"
            >
              <button
                class="button1 btn btn-secondary me-4 fs-4"
                type="button"
              >
                Sign Up
              </button>
            </a>
          </div>
          <p />
        </form>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
